chia/wallet/coin_selection.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import random
+from typing import Dict, List, Optional, Set
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint64, uint128
+from chia.wallet.wallet_coin_record import WalletCoinRecord
+
+
+async def select_coins(
+    spendable_amount: uint128,
+    max_coin_amount: uint128,
+    spendable_coins: List[WalletCoinRecord],
+    unconfirmed_removals: Dict[bytes32, Coin],
+    log: logging.Logger,
+    amount: uint128,
+    exclude: Optional[List[Coin]] = None,
+) -> Set[Coin]:
+    """
+    Returns a set of coins that can be used for generating a new transaction.
+    """
+    if exclude is None:
+        exclude = []
+
+    if amount > spendable_amount:
+        error_msg = (
+            f"Can't select amount higher than our spendable balance.  Amount: {amount}, spendable: {spendable_amount}"
+        )
+        log.warning(error_msg)
+        raise ValueError(error_msg)
+
+    log.info(f"About to select coins for amount {amount}")

```

Comment:
Maybe these and others should be Debug instead of Info?

Reply:
yep